.github/workflows/PRBuild.yaml

```diff
@@ -0,0 +1,111 @@
+name: CI
+
+on:
+  pull_request:
+    branches: [ master ]
+
+permissions:
+  contents: read
+
+defaults:
+  run:
+    shell: PowerShell
+
+jobs:
+  AnalyzeImages:
+    runs-on: [ windows-latest ]
+    outputs:
+      genericTag: ${{ steps.Analyze.outputs.genericTag }}
+      buildImagesJson: ${{ steps.Analyze.outputs.buildImagesJson }}
+      digestsJson: ${{ steps.Analyze.outputs.digestsJson }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Analyze
+        id: Analyze
+        run: |
+          $erroractionpreference = "STOP"
+          try {
```

Comment:
Suggest moving this into a PS file instead and pass in parameters fro $revision, etc, so this can be used locally.